File: sync-highlights.js
#!/usr/bin/env node
// sync-highlights.js
// Run: node sync-highlights.js
// Scans images/highlights/ and updates the highlights array in data/photos.json

const fs   = require('fs');
const path = require('path');

const ROOT          = __dirname;
const HIGHLIGHTS_DIR = path.join(ROOT, 'images', 'highlights');
const JSON_FILE      = path.join(ROOT, 'data', 'photos.json');
const IMAGE_EXTS     = new Set(['.jpg', '.jpeg', '.png', '.webp', '.JPG', '.JPEG', '.PNG', '.WEBP']);

const files = fs.readdirSync(HIGHLIGHTS_DIR)
  .filter(f => IMAGE_EXTS.has(path.extname(f)))
  .sort();

if (files.length === 0) {
  console.log('No images found in images/highlights/');
  process.exit(0);
}

const data = JSON.parse(fs.readFileSync(JSON_FILE, 'utf8'));

data.highlights = files.map(f => ({
  src:     `images/highlights/${f}`,
  caption: ''
}));

fs.writeFileSync(JSON_FILE, JSON.stringify(data, null, 2));

console.log(`✓ Synced ${files.length} image(s) to photos.json:`);
files.forEach(f => console.log(`  ${f}`));

File: sync-highlights.py
#!/usr/bin/env python3
"""
sync-highlights.py
Run: python3 sync-highlights.py
Scans images/highlights/ and updates the highlights array in data/photos.json
"""

import json, os, sys

ROOT           = os.path.dirname(os.path.abspath(__file__))
HIGHLIGHTS_DIR = os.path.join(ROOT, "images", "highlights")
JSON_FILE      = os.path.join(ROOT, "data", "photos.json")
IMAGE_EXTS     = {".jpg", ".jpeg", ".png", ".webp"}

files = sorted(
    f for f in os.listdir(HIGHLIGHTS_DIR)
    if os.path.splitext(f)[1].lower() in IMAGE_EXTS
)

if not files:
    print("No images found in images/highlights/")
    sys.exit(0)

with open(JSON_FILE, "r") as fh:
    data = json.load(fh)

data["highlights"] = [
    {"src": f"images/highlights/{f}", "caption": ""}
    for f in files
]

with open(JSON_FILE, "w") as fh:
    json.dump(data, fh, indent=2, ensure_ascii=False)

print(f"✓ Synced {len(files)} image(s) to photos.json:")
for f in files:
    print(f"  {f}")
